lib/service_actor/checkable.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module ServiceActor::Checkable
+  def self.included(base)
+    base.prepend(PrependedMethods)
+  end
+
+  module PrependedMethods
+    def _call
+      checks_for(:input)
+      super
+      checks_for(:output)
+    end
+
+    private
+
+    def checks_for(origin) # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+      self.class.public_send("#{origin}s").each do |input_key, input_options| # rubocop:disable Metrics/BlockLength
+        input_options.each do |check_name, check_conditions| # rubocop:disable Metrics/BlockLength
+          checks = %w[
+            TypeCheck
+            MustCheck
+            InclusionCheck
+            NilCheck
+            DefaultCheck
+          ]
+
+          checks.each do |check_class_name|
+            check_class =
+              Object.const_get("ServiceActor::Checks::#{check_class_name}")
+
+            argument_errors = check_class.check(
```

Comment:
If instead of `check` we called both `.new` and `#check` here we could remove the `.check` class method on checks.

Reply:
I have applied some changes based on your other comments. In my opinion, now the `self.check` layer is even more necessary. It accepts generic names and attributes, and then prepares specific attributes with specific names to work on the class instance.